v2/spiffetls/tlsconfig/config.go

```diff
@@ -8,63 +8,122 @@ import (
 	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
 )
 
+type verifyPeerCertificate = func(raw [][]byte, certs [][]*x509.Certificate) error
+
+func wrapVerifyPeerCertificate(wrapped, newVerifier verifyPeerCertificate) verifyPeerCertificate {
+	if wrapped == nil {
+		return newVerifier
+	}
+
+	return func(raw [][]byte, certs [][]*x509.Certificate) error {
+		if err := newVerifier(raw, certs); err != nil {
+			return err
+		}
+		return wrapped(raw, certs)
+	}
+}
+
 // TLSClientConfig returns a TLS configuration which verifies and authorizes
 // the server X509-SVID.
 func TLSClientConfig(svid x509svid.Source, bundle x509bundle.Source, authorizer Authorizer) *tls.Config {
-	panic("not implemented")
+	config := new(tls.Config)
+	HookTLSClientConfig(config, svid, bundle, authorizer)
+	return config
 }
 
 // HookTLSClientConfig sets up the TLS configuration to verify and authorize
 // the server X509-SVID. If there is an existing callback set for
 // VerifyPeerCertificate it will be wrapped by by this package and invoked
 // after SPIFFE authentication has completed.
 func HookTLSClientConfig(config *tls.Config, svid x509svid.Source, bundle x509bundle.Source, authorizer Authorizer) {
-	panic("not implemented")
+	config.InsecureSkipVerify = true
+	config.GetCertificate = GetCertificate(svid)
```

Comment:
Whoops, this is my bad. `TLSClient` functions should not be providing the client certificate.... we should get rid of the `x509svid.Source` parameter to these functions. Also, clients, when providing the certificate, use the `GetClientCertificate` callback, not `GetCertificate`.